projects/bp-gallery/src/components/views/geoguessr/GeoView.tsx

```diff
@@ -63,18 +68,29 @@ const GeoView = () => {
   const archives: FlatArchiveTag[] | undefined =
     useSimplifiedQueryResponseData(picturesData)?.archiveTags;
 
+  const isGeoCollectionPictures = useFlag('geopictures_from_collection');
+  const geoCollectionId = '357';
+  const { data: geoCollectionPictureData } = useGetPicturesForCollectionQuery({
+    variables: { collectionId: geoCollectionId },
+  });
+  const geoCollectionPictureIds: string[] | undefined = useSimplifiedQueryResponseData(
+    geoCollectionPictureData
+  )?.collection?.pictures?.map(picture => picture?.id ?? '');
+
   useEffect(() => {
     if (!archives) {
       return;
     }
-    const allPictureIds = getAllPictureIds(archives);
+    const allPictureIds = isGeoCollectionPictures
+      ? geoCollectionPictureIds ?? ['3']
+      : getAllPictureIds(archives);
     const shuffledPictureIds = shufflePictureIds(allPictureIds, seed);
     pictureQueue.current = getTodaysPictureQueue(shuffledPictureIds);
     setPictureId(getNextPicture());
-  }, [archives]);
+  }, [archives, geoCollectionPictureIds, isGeoCollectionPictures]);
 
   const getNextPicture = () => {
-    return pictureQueue.current.shift() ?? fallbackPictureId;
+    return pictureQueue.current.shift() ?? '';
```

Comment:
I think using something like `null` is more explicit than an empty string. It also forces you to handle the case that an archive doesn't have any pictures, which is currently unhandled in the `useEffect` above, I think.